pkg/daemon/ceph/client/pool.go

```diff
@@ -190,21 +189,28 @@ func CreateECPoolForApp(context *clusterd.Context, clusterName string, newPool C
 
 	buf, err := ExecuteCephCommand(context, clusterName, args)
 	if err != nil {
+		DeleteErasureCodeProfile(context, clusterName, newPool.ErasureCodeProfile)
```

Comment:
Since the Rook operator is idempotent, every time the operator starts it will ensure the resources are created, so this may be called even after the initial creation. So we may need to differentiate between new creation where it is safe to delete resources and an existing pool where we wouldn't want to clean up. For example, perhaps we need to check for existence before creating it, then skip creating it if it already exists.

Reply:
sounds right.